docs/pages/docs/model-based-testing.mdx

```diff
@@ -1,9 +1,213 @@
-import { Callout } from 'nextra/components'
+import { Callout, Steps } from 'nextra/components'
 
 # Model-based Testing
 
-<Callout type="warning" emoji="🚧️">
-  This page is under construction.
-</Callout>
+One of the most asked questions when it comes to writing specifications is "what about the code?". Or, in a more expressive problem statement:
+> How to bridge the gap between the specification and the implementation?
 
-For now, refer to Informal System's page on Model-Based Techniques: [mbt.informal.systems](https://mbt.informal.systems/)
+That is, even if we verify that some model is correct, there could still be problems in the code that is ran in production, which is what actually matters. This specially likely to happen if the specification is burried deep into some project sub-sub-sub-folder and no one dares to open it, let alone update it as the code evolves.
+
+The perfect solution would be: verify your code! But that is far from easy, and can most likely only be accomplished with a fat budget (or some really smart volunteers). Other people might say: write the spec in several layers of abstraction until you are really close to the implementation, proving the refinement between each of them - but that is also hard, and doesn't even address the problem of keeping code and spec in sync. Finally, there's code generation (or extracting code from the formal spec), which is great for some sequential algorithms, but get's weird in many ways when you start having multiple processes - and that discussion doesn't fit in this page.
+
+Let's focus on more practical solutions:
+  1. Model-based Testing (MBT)
+  2. Trace validation
+
+Both of these won't give you a proof that your code is correct, but they will make you more confident that the behavior between the model (specification) and the code (implementation) matches, even throughout changes to the code base (or the spec, which tends to change way less).
+
+## MBT in one paragraph
+
+Produce a bunch of traces (executions) from your model, which should be valid traces in your system. For example: Bob deposits $100 and then transfers $60 to Alice, and then Alice withdraws $20. A trace is just a JSON file with all the data you need. You then write a test driver to call the respective functions (`deposit("Bob", 100)`, `transfer("Bob", "Alice", 60)`, `withdraw("Alice", 20)`) in your implementation, and optionally check things between each call or at the very end (such as asserting a property or comparing the state). That's it, you use the model to generate test data (inputs and expectations) for you in a very efficient way.
+
+| Data  | Expectations |
+| :---  | :---         |
+| Model | Model        |
+
+
+## Trace validation in one paragraph
```

Comment:
Is there more coming? We don't have a detailed section on this. We should at least mention that there is none, if there is none.

Reply:
I added this

<img width="1461" height="832" alt="image" src="https://github.com/user-attachments/assets/e77af7ee-48d2-4927-9a0c-b6259660596f" />